pkg/shim/scheduler_test.go

```diff
@@ -132,7 +132,7 @@ partitions:
 
 	// wait for scheduling app and tasks
 	// verify app state
-	cluster.waitAndAssertApplicationState(t, appID, events.States().Application.Failed)
+	cluster.waitAndAssertApplicationState(t, appID, events.States().Application.Failing)
```

Comment:
I don't think this fix is OK here. When the application is rejected by the scheduler, the app should fail. Otherwise in this scenario, the app will never Fail, it will stuck in the Failing state.

Reply:
Yes, correct. In case of app rejection, we should directly move the app to 'Failed' state. Handled it using appropriate state transition, rejected -> failed during fail application event. Please check.